### PR TITLE
Trace logs for base node response with wallet UTXO queries

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
Added additional trace logs to enable fault finding unsuccessful queries to be handled between the base node wallet and the base node.

## Motivation and Context
Assist in fault finding as messages uncovering the root cause cannot be traced

## How Has This Been Tested?
Tested with a base node

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/core/src/base_node/service/service.rs [hunk 1]
```diff
@@ -407,13 +407,68 @@ async fn handle_incoming_request<B: BlockchainBackend + 'static>(
...
+    );
+    match outbound_message_service
         .send_direct(
             origin_public_key,
             OutboundEncryption::None,
             OutboundDomainMessage::new(TariMessageType::BaseNodeResponse, message),
         )
-        .await?;
+        .await
+    {
```
Comment (philipr-za) [inline base_layer/core/src/base_node/service/service.rs]: So adding this in does give more detail in terms of the send process but it also might have some unintended consequences.

The prior code started the send and then moved on. This added code adds the `result.resolve_ok().await` which will wait for discovery to complete if it needs to and `send_states.wait_single().await ` will wait for the send to pass or fail which can take time depending on the Tor situation.

So I just want to make sure that this potential extra time (up to 2 minutes for discovery and up to 6 minutes by default on the Tor connection deciding that the destination is unreachable) will not derail other tasks within the BN?

Comment (hansieodendaal) [reply base_layer/core/src/base_node/service/service.rs]: The way I understand is that `if send_states.len() == 1` the next `send_states.wait_single().await` follows directly. We could hav a look at the timing in the logs, with and without send errors.

Comment (hansieodendaal) [reply base_layer/core/src/base_node/service/service.rs]: Ultimately a call to `send_direct` calls `pub async fn send_raw`, which will wait for a reply message from the receiver over the wire. This is where all the significant waiting will happen if any. The handling of message states afterwards is internal and quick.

Comment (philipr-za) [reply base_layer/core/src/base_node/service/service.rs]: So the first thing that can take long is that you have added the `result.resolve_ok().await`. If discovery has to happen then that await can wait for multiple minutes. If we don't need to do discovery then the message is queued to be sent.

Once the message is queued we wait at `send_states.wait_single().await`. This explicitly waits until the queued message is sent or fails to send. Usually this takes about 8 seconds BUT if the other party is not reachable we have seen tor take up to 6-8 minutes to decide that the other party cannot be reached.

With the code you are replacing all of the discovery and waiting for the queued message send is not waited on in this code path. It treats the send as a fire and forget and lets the comms stack handle it. By adding this code in you are explicitly waiting on the various stages of sending. So the question is: Can this code tolerate those long waits?

Comment (hansieodendaal) [reply base_layer/core/src/base_node/service/service.rs]: Understood, let's wait then before continuing. Need some more insight into this issue.

Comment (hansieodendaal) [reply base_layer/core/src/base_node/service/service.rs]: All added log messages dependent on `send_states. ... .await` or `.resolve_ok().await` are now executed from within a `tokio::spawn(async move` task, so no blocking calls anymore.

**Review by philipr-za — APPROVED** [review]
LGTM

**Pull request merged**